File: source_funcs/go_transcoder/K_TH_DIGIT_RAISED_POWER_B/K_TH_DIGIT_RAISED_POWER_B.go
package main

import "math"

func f_gold(a int, b int, k int) int {
	var (
		p     int = int(math.Pow(float64(a), float64(b)))
		count int = 0
	)
	for p > 0 && count < k {
		var rem int = p % 10
		count++
		if count == k {
			return rem
		}
		p = p / 10
	}
	return 0
}

File: rust_funcs/c_transcoder/K_TH_DIGIT_RAISED_POWER_B/K_TH_DIGIT_RAISED_POWER_B.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

fn f_gold(a: i32, b: i32, k: i32) -> i32 {
    let mut p = a.pow(b);
    let mut count = 0;
    while p > 0 && count < k {
        let rem = (p % 10) as i32;
        count += 1;
        if count == k {
            return rem;
        }
        p = p / 10;
    }
    return 0;
}
